geode-core/src/main/java/org/apache/geode/management/internal/JmxManagerLocator.java

```diff
@@ -203,7 +202,7 @@ private boolean sendStartJmxManager(InternalDistributedMember distributedMember)
     @Override
     public void execute(FunctionContext context) {
       try {
-        Cache cache = CacheFactory.getAnyInstance();
+        InternalCache cache = ManagementAgent.getCache();
```

Comment:
I think this should probably change to:
```
 InternalCache cache = toInternalCacheForClientAccess(context.getCache());
```
Then you could replace ManagementAgent.getCache() with:
```
public static InternalCacheForClientAccess toInternalCacheForClientAccess(Cache cache) {
  return ((InternalCache)cache).getCacheForProcessingClientRequests();
}
```
Or if you think FunctionContext.getCache() should always return InternalCacheForClientAccess, you could move that code into FunctionContext.